### PR TITLE
Conditionally run namedPipes tests in uap if windows version contains fix

### PR DESCRIPTION
**Pull request by joperezr**
Cc @danmosemsft @joshfree @safern 

Adding tests back conditionally for new Windows version. This doesn't close the related issue yet, because there are two more tests still failing (pingpong_async and PingPong_sync). I'll continue investigating those to see why did we get that regression.

### CODE REVIEW

**Comment by joperezr** [comment]
Adding no merge label until I send out the ask mode mail and get an approval.

**Comment by joshfree** [comment]
> Can we get this into master?

Yes, please put this in master.

File: src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs [hunk 1]
```diff
@@ -28,9 +30,11 @@ protected override ServerClientPair CreateServerClientPair()
         }
     }
     
-    [ActiveIssue(22271, TargetFrameworkMonikers.UapNotUapAot)]
+
```
Comment (safern) [inline src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs]: Nit => extra line.

File: src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs [hunk 2]
```diff
@@ -8,9 +8,11 @@
 
 namespace System.IO.Pipes.Tests
 {
-    [ActiveIssue(22271, TargetFrameworkMonikers.UapNotUapAot)]
+
```
Comment (safern) [inline src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs]: Nit => extra line.

File: src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs [hunk 3]
```diff
@@ -48,9 +52,11 @@ protected override ServerClientPair CreateServerClientPair()
         }
     }
     
-    [ActiveIssue(22271, TargetFrameworkMonikers.UapNotUapAot)]
+
```
Comment (safern) [inline src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Write.cs]: Nit => extra line.

File: src/System.IO.Pipes/tests/PipeTest.Read.cs [hunk 4]
```diff
@@ -16,193 +16,223 @@ namespace System.IO.Pipes.Tests
     /// </summary>
     public abstract partial class PipeTest_Read : PipeTestBase
     {
+
```
Comment (safern) [inline src/System.IO.Pipes/tests/PipeTest.Read.cs]: Nit => Extra line.

File: src/System.IO.Pipes/tests/PipeTest.Write.cs [hunk 5]
```diff
@@ -14,214 +14,246 @@ namespace System.IO.Pipes.Tests
     /// </summary>
     public abstract class PipeTest_Write : PipeTestBase
     {
+        protected virtual bool ShouldSkipTest() => false;
```
Comment (safern) [inline src/System.IO.Pipes/tests/PipeTest.Write.cs]: Nit => You could add this to PipeTestBase instead and have only one instance of 
```cs
protected virtual bool ShouldSkipTest() => false;
```

**Comment by safern** [comment]
> @safern I'm assuming this is indeed the cleanest way to do it.

Given that ConditionalFact and ConditionalTheory demand that the property is static yes :(

File: src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Read.cs [hunk 6]
```diff
@@ -8,9 +8,10 @@
 
 namespace System.IO.Pipes.Tests
 {
-    [ActiveIssue(22271, TargetFrameworkMonikers.UapNotUapAot)]
+
```
Comment (danmosemsft) [inline src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Read.cs]: nit, blank line :)

File: src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Read.cs [hunk 7]
```diff
@@ -8,9 +8,10 @@
 
 namespace System.IO.Pipes.Tests
 {
-    [ActiveIssue(22271, TargetFrameworkMonikers.UapNotUapAot)]
+    
     public class NamedPipeTest_Read_ServerOut_ClientIn : PipeTest_Read
     {
+        protected override bool ShouldSkipTest() => PlatformDetection.IsWinRT && !PlatformDetection.IsWindows10Version16256OrGreater;
```
Comment (danmosemsft) [inline src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Read.cs]: I can't find where ShouldSkipTest is consumed?

Comment (danmosemsft) [reply src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Read.cs]: Oh nm diff wasn't expanded

File: src/System.IO.Pipes/tests/PipeTest.Write.cs [hunk 8]
```diff
@@ -14,214 +14,246 @@ namespace System.IO.Pipes.Tests
...
+
         [Fact]
         public void WriteWithNullBuffer_Throws_ArgumentNullException()
         {
-            using (ServerClientPair pair = CreateServerClientPair())
+            if (!ShouldSkipTest())
```
Comment (danmosemsft) [inline src/System.IO.Pipes/tests/PipeTest.Write.cs]: No need to change now, but in general, I prefer the style of 
```c#
public void WriteWithNullBuffer_Throws_ArgumentNullException()
{
        if (ShouldSkipTest())
               return;
   ...
```

ie., where an if wraps the whole body of the method, return early instead. This avoids "triangle code" and in this case also avoids a double negative which are often a source of bugs.

Comment (danmosemsft) [reply src/System.IO.Pipes/tests/PipeTest.Write.cs]: It also would greatly reduce the diff :)

**Comment by danmosemsft** [comment]
Can you please put this into master first.

**Comment by joshfree** [comment]
updating milestone to 6.1.  @joperezr can you merge this into master first?

**Comment by joperezr** [comment]
That was done a couple of weeks ago in master: https://github.com/dotnet/corefx/pull/24307

We did however have to make some changes afterwards as some tests would still hang on RS3 machines.

**Comment by karelz** [comment]
@joshfree is the branch release/uwp6.0 repurposed for UWP6.1 work?

**Comment by joshfree** [comment]
@karelz release/uwp6.0 is for 6.0.x servicing.  6.1 should just go to master for now.  

/cc @russellhadley who is running the UWP next release and may wish to alter the branching strategy going forward.

**Comment by danmosemsft** [comment]
If we are getting test hangs in release/uwp6.0 we should consider porting test-only changes there just for the sake of having a stable branch to service from. Otherwise I agree we should not touch the branch.

**Comment by karelz** [comment]
Per above guidance: Changing the milestone to UWP6.0.x (from UWP6.1) as it is against release/uwp6.0 branch ...

**Comment by karelz** [comment]
@joperezr what is status of this PR?

**Comment by karelz** [comment]
@joperezr ping?

**Comment by joperezr** [comment]
This PR wasn't merged before because shiproom wanted me to hold off on it until after release. Now that this branch will serve for servicing we can go ahead and merge it. Before that though, I'll re-run CI since it is a fairly old PR so CI might not be green any longer.

**Comment by joperezr** [comment]
@dotnet-bot test this please

**Pull request closed without merging**